scripts/translate_doc_pr/pr_analyzer.py

```
@@ -0,0 +1,1447 @@
+                        'title': title,
+                        'line': line
+                    }
+        
+        # Build complete hierarchy from HEAD (after changes)
+        all_hierarchy_dict = build_hierarchy_dict(file_content)
+        
+        # For deletion detection, we also need the base file hierarchy
+        try:
+            base_file_content = repository.get_contents(file.filename, ref=f"{repository.default_branch}").decoded_content.decode('utf-8')
+            base_hierarchy_dict = build_hierarchy_dict(base_file_content)
+        except Exception as e:
```

Ruff (0.13.1)

Do not catch blind exception: `Exception`

(BLE001)